department-of-veterans-affairs/abd-vro
# MCP 2032: Cancer processor

## What was the problem?

No support for Cancer condition.

Associated tickets or Slack threads:

- [MCP-2032](https://amida.atlassian.net/browse/MCP-2032)

## How does this fix it?[^1]

Adds a new processor for cancer condition, with code sets. 
There is no mechanism in place to route multiple conditions or VASRD codes to a single processor so for now it is hard-coded to use 'respiratory' cancer code sets when filtering medication and conditions. 

## How to test this PR
- Step 1 
 `./gradlew -PenablePrototype build check docker`                    
`export COMPOSE_PROFILES=prototype`
`./gradlew :app:dockerComposeDown :app:dcPrune :app:dockerComposeUp`

- Step 2 Put "cancer" as diagnostic code, ICN 38000035
- [^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
